tests/unit/helper.ts

```diff
@@ -13,7 +14,15 @@ export const wallet = {
   publicKey: "0xea526ba1710343d953461ff68641f1b7df5f23b9042ffa2d2a798d3adb3f3d6c",
 };
 
-/* eslint-disable max-len */
+export const secp256k1WalletTestObject = {
```

Comment:
Can you document where you got this test case from?

Also, what about the SLIP-0010 test vectors? Why not add one of those as well?

Reply:
it is a bit messy because of the regex validation check we do 
```
APTOS_BIP44_REGEX = /^m\/44'\/637'\/[0-9]+'\/[0-9]+\/[0-9]+$/;
```

Reply:
Ah, I see.

Reply:
I was just using known mnemonic with this path to generate an account

Reply:
Sorry, so how is the test case generated?

Reply:
I used the `Account.fromDerivationPath()` with 
```
mnemonic: "shoot island position soft burden budget tooth cruel issue economy destroy above",
path: "m/44'/637'/0'/0/0",
```
and used the privateKey, publicKey and address it generated

Reply:
I see. I worry that we don't have test cases generated with other libraries. I guess it's not as easy to add those because our code harcodes `637`.

Reply:
yes.

Reply:
I would advise we find some generic BIP-44+32 library, generate a test case using it and then add it as a test here, to make sure we are compatible.

Otherwise, if we get this wrong, when hardware wallets try to support our (incorrect) BIP-44+32 implementation, we will have issues.

Reply:
I guess we can test directly against https://github.com/aptos-labs/aptos-ts-sdk/pull/146/files#diff-98942ba717d1ba7eafd0b4cfa834f6307d611d3ba525aa2ae68a93938b97e0b5R189
```
const { privateKey } = HDKey.fromMasterSeed(mnemonicToSeed(mnemonics)).derive(path);
```
to bypass the validation check, this is what I did in my local tests. is this enough? 
if so, I can do it in a separate PR since we need this PR in to unblock work

Reply:
That would defeat the point of testing, no? We are testing the code against itself, AFAICT.

We would need externally-computed private keys from some mnemonic + path.

But, sure, you can do it in another PR. 

After all, it looks like you are just directly using the scure/bip44 library, except for the `mnemonicToSeed` function, which cleans the input before calling scure/bip39. So, unless there's something wrong in our `mnemonicToSeed`, there shouldn't be any compatibility issues.

You could also just move this `HDKey.fromMasterSeed(mnemonicToSeed(mnemonics)).derive(path)` line into its own `fromDerivationPathInternal` function, and then test it against the SLIP-0010 vectors, avoiding the Aptos-specific path checks.